### PR TITLE
Added an option to show a search button and an clear filter button

### LINKED ISSUE #3994: Table data are not refreshed when clear one search text then clear another one.
We make a ui as snapshot. Table data are not refreshed when clear one search text and then clear another one. 
image
I found that on that occasion "onSearch" event is triggered but because of "text === this.searchText" (both blank), function is returned, refreshing is not executed.

    onSearch ({currentTarget, firedByInitSearchText}) {
     .........
      if (text === this.searchText) {
        return
      }
    .............
    }

Reproduce steps
1.  input search text in one filter control (refresh)
2. input search text in search text (refresh)
3. clear all search text in the filter control one time(refresh)
4. clear all search text in search text (not refresh)

### LINKED ISSUE #4194: Submit and clear buttons for search field
On my project I wanted the search field for each table to have optional dedicated submit and clear buttons and I figured I'd see if this would be a wanted feature.

[Here is my initial update](https://github.com/doug-the-guy/bootstrap-table/commit/f0612d8378879126cecc4477868e59163377ef10)

This happened to coincide with your big theme overhaul. I'm not sure if it will be possible to generalize the HTML syntax for all of the these because each UI framework constructs the input-group differently. So far I've verified with Bootstrap 4 but have coded up the configurations for Bootstrap 3 and Bulma. I had to take a little liberties so far (i.e. a useless div around the input search field only because Bulma needs it). Before I continue integrating the other UI frameworks let me know if you'd think this would be a good addition. If so, I'll put up some jsFiddles to demonstrate the new functionality.

Here is how the search field will be rendered in the three currently implemented UI frameworks:

```
<!-- bulma -->
<div class="float-right search">
    <div class="field has-addons">
        <div class="control">
            <input class="input" type="text" placeholder="Search">
        </div>
        <div class="control">
            <button type="button" class="button" name="search" aria-label="Search" title="Submit Search"><i class="fa fa-search"></i></button>
        </div>
        <div class="control">
            <button type="button" class="button" name="clearSearch" aria-label="Clear Search" title="Clear Search"><i class="fa fa-times"></i></button>
        </div>
    </div>
</div>

<!-- bootstrap 3 -->
<div class="float-right search">
    <div class="input-group">
        <div class="">
            <input class="form-control" type="text" placeholder="Search">
        </div>
        <div class="input-group-btn">
            <button class="btn btn-secondary" type="button" name="search" aria-label="Search" title="Submit Search"><i class="glyphicon glyphicon-search"></i></button>
        </div>
        <div class="input-group-btn">
            <button class="btn btn-secondary" type="button" name="clearSearch" aria-label="Clear Search" title="Clear Search"><i class="glyphicon glyphicon-remove"></i></button>
        </div>
    </div>
</div>

<!-- bootstrap 4 -->
<div class="float-right search">
    <div class="input-group">
        <div class="">
            <input class="form-control" type="text" placeholder="Search">
        </div>
        <div class="input-group-append">
            <button class="btn btn-secondary" type="button" name="search" aria-label="Search" title="Submit Search"><i class="fa fa-search"></i></button>
        </div>
        <div class="input-group-append">
            <button class="btn btn-secondary" type="button" name="clearSearch" aria-label="Clear Search" title="Clear Search"><i class="fa fa-times"></i></button>
        </div>
    </div>
</div>
```

These options can be enabled with boolean fields 'searchButton' and 'clearSearchButton'.

I think it would be best to make these attached to the search field (so it is clear they operate on the it and not the table like the other toolbar buttons) but if it can't be done generally for every theme then they can just be attached to the general toolbar if 'search' is enabled.

### PR DESCRIPTION
**Pull request by UtechtDustin**
fix #4194
will also fix #3994 

Example: http://jsfiddle.net/myp2tzdb/6/

### CODE REVIEW

File: src/locale/bootstrap-table-es-AR.js [hunk 1]
```diff
@@ -20,6 +20,9 @@ $.fn.bootstrapTable.locales['es-AR'] = {
   formatDetailPagination (totalRows) {
     return `Showing ${totalRows} rows`
   },
+  formatClearSearch () {
+    return 'Clear Search'
```
Comment (djhvscf) [inline src/locale/bootstrap-table-es-AR.js]: Let's use 'Limpiar búsqueda'

File: src/locale/bootstrap-table-es-CL.js [hunk 2]
```diff
@@ -21,6 +21,9 @@ $.fn.bootstrapTable.locales['es-CL'] = {
   formatDetailPagination (totalRows) {
     return `Showing ${totalRows} rows`
   },
+  formatClearSearch () {
+    return 'Clear Search'
```
Comment (djhvscf) [inline src/locale/bootstrap-table-es-CL.js]: Let's use 'Limpiar búsqueda'

File: src/locale/bootstrap-table-es-CR.js [hunk 3]
```diff
@@ -20,6 +20,9 @@ $.fn.bootstrapTable.locales['es-CR'] = {
   formatDetailPagination (totalRows) {
     return `Showing ${totalRows} rows`
   },
+  formatClearSearch () {
+    return 'Clear Search'
```
Comment (djhvscf) [inline src/locale/bootstrap-table-es-CR.js]: Let's use 'Limpiar búsqueda'

File: src/locale/bootstrap-table-es-ES.js [hunk 4]
```diff
@@ -20,6 +20,9 @@ $.fn.bootstrapTable.locales['es-ES'] = {
   formatDetailPagination (totalRows) {
     return `Showing ${totalRows} rows`
   },
+  formatClearSearch () {
+    return 'Borrar los filtros'
```
Comment (djhvscf) [inline src/locale/bootstrap-table-es-ES.js]: Let's use 'Limpiar búsqueda'

File: src/locale/bootstrap-table-es-MX.js [hunk 5]
```diff
@@ -22,6 +22,9 @@ $.fn.bootstrapTable.locales['es-MX'] = {
   formatDetailPagination (totalRows) {
     return `Mostrando ${totalRows} filas`
   },
+  formatClearSearch () {
+    return 'Clear Search'
```
Comment (djhvscf) [inline src/locale/bootstrap-table-es-MX.js]: Let's use 'Limpiar búsqueda'

File: src/locale/bootstrap-table-es-NI.js [hunk 6]
```diff
@@ -20,6 +20,9 @@ $.fn.bootstrapTable.locales['es-NI'] = {
   formatDetailPagination (totalRows) {
     return `Showing ${totalRows} rows`
   },
+  formatClearSearch () {
+    return 'Clear Search'
```
Comment (djhvscf) [inline src/locale/bootstrap-table-es-NI.js]: Let's use 'Limpiar búsqueda'

File: src/locale/bootstrap-table-es-SP.js [hunk 7]
```diff
@@ -20,6 +20,9 @@ $.fn.bootstrapTable.locales['es-SP'] = {
   formatDetailPagination (totalRows) {
     return `Showing ${totalRows} rows`
   },
+  formatClearSearch () {
+    return 'Clear Search'
```
Comment (djhvscf) [inline src/locale/bootstrap-table-es-SP.js]: Let's use 'Limpiar búsqueda'

File: src/locale/bootstrap-table-es-AR.js (hunk 1 above)
Comment (UtechtDustin) [reply src/locale/bootstrap-table-es-AR.js]: Already commit by u (thanks!), so i resolve the conversations :)

**Pull request merged**